### PR TITLE
LibGfx: Support color schemes

### PR DESCRIPTION
**Pull request by implicitfield**
Like #16174, but much better integrated with the system.

This allows themes to specify color schemes, which can then be used by programs through ```LibGfx```.
Currently only ```Flood``` and ```TerminalWidget``` support color schemes, but this makes adding such support to other programs much easier.

Feel free to comment if a color scheme other than the predefined one would be a better fit for the respective theme.

### CODE REVIEW

File: Ports/serenity-theming/package.sh [hunk 1]
```diff
@@ -21,7 +21,8 @@ post_install() {
     cp -r "${workdir}/fonts" "${SERENITY_INSTALL_ROOT}/res/"
     cp -r "${workdir}/color-palettes" "${SERENITY_INSTALL_ROOT}/res/"
     cp -r "${workdir}/emoji" "${SERENITY_INSTALL_ROOT}/res/"
-    cp -r "${workdir}/terminal-colors" "${SERENITY_INSTALL_ROOT}/res/"
+    # FIXME: Update this path upstream.
```
Comment (davidot) [inline Ports/serenity-theming/package.sh:24]: Feel free to also make a PR on the theming repo with this change (with a link to this PR)

File: Userland/Games/Flood/BoardWidget.cpp [hunk 2]
```diff
@@ -8,12 +8,30 @@
...
+    Vector<Color> color_scheme {
+        palette.bright_black(),
+        palette.bright_red(),
+        palette.bright_green(),
+        palette.bright_yellow(),
+        palette.bright_blue(),
+        palette.bright_magenta(),
+        palette.bright_cyan(),
+        palette.bright_white()
+    };
```
Comment (davidot) [inline Userland/Games/Flood/BoardWidget.cpp:31]: You can probably inline the construction of the vector into the `set_color_scheme` call

File: Userland/Libraries/LibVT/TerminalWidget.cpp [hunk 3]
```diff
@@ -197,6 +196,9 @@ void TerminalWidget::focusout_event(GUI::FocusEvent& event)
 
 void TerminalWidget::event(Core::Event& event)
 {
+    if (event.type() == GUI::Event::ThemeChange)
+        update_color_scheme();
+
```
Comment (davidot) [inline Userland/Libraries/LibVT/TerminalWidget.cpp:201]: You can chain this with the following ifs (with an else if)

File: Userland/Libraries/LibGfx/SystemTheme.cpp [hunk 4]
```diff
@@ -40,8 +40,23 @@ Core::AnonymousBuffer load_system_theme(Core::ConfigFile const& file)
...
         auto color_string = file.read_entry("Colors", name);
         auto color = Color::from_string(color_string);
-        if (!color.has_value())
-            return Color(Color::Black);
+        if (!color.has_value()) {
+            auto maybe_color_config = Core::ConfigFile::open(data->path[(int)PathRole::ColorScheme]);
+            if (maybe_color_config.is_error())
+                maybe_color_config = Core::ConfigFile::open("/res/color-schemes/Default.ini");
+            auto color_config = maybe_color_config.release_value();
+            if (strcmp(name, "ColorSchemeBackground") == 0)
```
Comment (davidot) [inline Userland/Libraries/LibGfx/SystemTheme.cpp:48]: Why use strcmp here? Can't we just use `==` against a stringview?
`if (name == "ColorSchemeBackground"sv)`

File: Userland/Libraries/LibGfx/SystemTheme.cpp [hunk 5]
```diff
@@ -40,8 +40,23 @@ Core::AnonymousBuffer load_system_theme(Core::ConfigFile const& file)
...
+                color = Gfx::Color::from_string(color_config->read_entry("Primary", "Background"));
+            else if (strcmp(name, "ColorSchemeForeground") == 0)
+                color = Gfx::Color::from_string(color_config->read_entry("Primary", "Foreground"));
+            else if (strncmp(name, "Bright", 6) == 0)
```
Comment (davidot) [inline Userland/Libraries/LibGfx/SystemTheme.cpp:52]: And `.starts_with` here?

File: Userland/Libraries/LibGfx/SystemTheme.h [hunk 6]
```diff
@@ -31,6 +31,24 @@ namespace Gfx {
...
+    C(Cyan)                        \
+    C(White)                       \
+    C(BrightBlack)                 \
+    C(BrightRed)                   \
+    C(BrightGreen)                 \
+    C(BrightYellow)                \
+    C(BrightBlue)                  \
+    C(BrightMagenta)               \
+    C(BrightCyan)                  \
+    C(BrightWhite)                 \
```
Comment (davidot) [inline Userland/Libraries/LibGfx/SystemTheme.h:51]: The rest of this macro looks sorted alphabetically, so maybe also sort these?
Or at least put all these at the bottom/top to indicate they are some group

**Review by davidot — CHANGES_REQUESTED** [review]
The code looks fine to me, just a small couple of comments.

And one suggestion (which could also be solved in a later PR of course) is that one might want different color schemes for terminal and flood etc. Or at least some easy way to change it from within the app.

File: Userland/Libraries/LibGfx/SystemTheme.cpp (hunk 5 above)
Comment (davidot) [reply Userland/Libraries/LibGfx/SystemTheme.cpp:52]: Oh I didn't see that, hmm that's too bad then this is fine I guess

**Review by davidot — APPROVED** [review]
Code looks fine to me now! Nice work

**Comment by linusg** [comment]
The formatter is not happy

**Pull request merged**